packages/domain/lib/usecase/get_movie_list_usecase.dart

```diff
@@ -1,37 +1,101 @@
-    await _getMovies(itemCount, jsonMovies, movieType);
-    return jsonMovies;
+  Future<List<MovieCache>> call(MovieType currentMovieType) async {
+    final List<MovieCache> jsonMovies = [];
+    final List<MovieCache> jsonMoviesFromCache = [];
+    final List<int> movieCacheIds = [];
+    if (await _isTimestampNewDay(
+      DateTime.now(),
+      currentMovieType,
+    )) {
+      final GetMovieDataResponse response =
+          currentMovieType == MovieType.anticipated
+              ? await _repository.getMovieAnticipatedData()
+              : await _repository.getMovieTrendingData();
+      final pageCount = int.parse(response.headers[Config.pageCount][0]);
+      final itemCount = pageCount >= 5
+          ? 50
+          : int.parse(response.headers[Config.itemCount][0]);
+      await _getMoviesFromTrakt(
+        itemCount,
+        jsonMovies,
+        currentMovieType,
+      );
+      movieCacheIds.addAll(await _localStorageRepository
+          .getMoviesIdsFromCache(currentMovieType));
+      final List<int> idsToDeleteFromDB = movieCacheIds
+          .where((e1) => jsonMovies.where((e) => e.trakt == e1).isEmpty)
+          .toList();
+      await _localStorageRepository.deleteByIds(
+          idsToDeleteFromDB, currentMovieType);
+      final List<MovieCache> moviesToAddToDB = jsonMovies
+          .where(
+              (movie) => movieCacheIds.where((id) => movie.trakt == id).isEmpty)
+          .toList();
+      await _localStorageRepository.saveMoviesToCache(
+          moviesToAddToDB, currentMovieType);
+      return jsonMovies;
+    } else {
+      jsonMoviesFromCache.addAll(
+          await _localStorageRepository.getMoviesFromCache(currentMovieType));
+      return jsonMoviesFromCache;
+    }
   }
 
-  Future<List<MovieResponse>> _getMovies(
+  Future<List<MovieCache>> _getMoviesFromTrakt(
     int itemCount,
-    List<MovieResponse> jsonMovies,
+    List<MovieCache> jsonMovies,
     MovieType movieType,
   ) async {
     final response = (movieType == MovieType.anticipated)
         ? await _repository.getMovieAnticipatedData(itemCount: itemCount)
         : await _repository.getMovieTrendingData(itemCount: itemCount);
-    jsonMovies.addAll(response.body.map((e) => MovieResponse.fromJson(e)));
+    jsonMovies
+        .addAll(response.body.map((e) => MovieCache.fromResponse(e['movie'])));
     return jsonMovies;
   }
+
+  Future<bool> _isTimestampNewDay(
+    DateTime now,
+    MovieType movieType,
+  ) async {
+    final today = DateTime(now.year, now.month, now.day);
+    final int todayMilliseconds = today.millisecondsSinceEpoch;
+    final int? timestampFromCache =
+        await _localStorageRepository.getTimestampForMovieType(movieType);
+    if (timestampFromCache == null) {
+      _localStorageRepository.setTimestampForMovieType(
+        movieType,
+        now.millisecondsSinceEpoch,
+      );
+      return true;
+    } else if (timestampFromCache >= todayMilliseconds) {
```

Comment:
I think we can use simple if instead of else if as every if statements returns and exit method execution.